python_modules/dagster/dagster/_core/storage/cloud_storage_compute_log_manager.py

```diff
@@ -62,10 +95,31 @@ def cloud_storage_has_logs(
         """Returns whether the cloud storage contains logs for a given log key."""
 
     @abstractmethod
+    def _upload_file_obj(
+        self, data: IO[bytes], log_key: Sequence[str], io_type: ComputeIOType, partial=False
+    ):
+        pass
+
     def upload_to_cloud_storage(
         self, log_key: Sequence[str], io_type: ComputeIOType, partial: bool = False
     ) -> None:
         """Uploads the logs for a given log key from local storage to cloud storage."""
+        # We've already truncated
+        if (tuple(log_key), io_type) in self._truncated:
+            return
+
+        path = self.local_manager.get_captured_local_path(log_key, IO_TYPE_EXTENSION[io_type])
+        ensure_file(path)
+
+        max_bytes = int(os.environ.get("DAGSTER_TRUNCATE_COMPUTE_LOGS_UPLOAD_BYTES", "0"))
```

Comment:
we should not default this to 0 i think, there should be a reasonable limit (50MB?) by default

alternately you could make the default a property and we could override the default in the cloud compute log manager